lib/components/admin/call-taker-controls.js

```diff
@@ -64,11 +65,21 @@ class CallTakerControls extends Component {
     )
   }
 
-  _onToggleCallHistory = () => this.props.toggleCallHistory()
+  _onToggleCallHistory = () => {
+    const {callTaker, resetForm, toggleCallHistory, toggleFieldTrips} = this.props
+    resetForm(true)
+    toggleCallHistory()
+    if (callTaker.fieldTrip.visible) toggleFieldTrips()
+  }
 
-  _onToggleFieldTrips = () => this.props.toggleFieldTrips()
+  _onToggleFieldTrips = () => {
+    const {callTaker, resetForm, toggleCallHistory, toggleFieldTrips} = this.props
+    resetForm(true)
+    toggleFieldTrips()
+    if (callTaker.callHistory.visible) toggleCallHistory()
+  }
```

Comment:
Since none of these methods do anything with the component state, it'd make this component cleaner to put all of this logic into an action file.

Reply:
Added in https://github.com/opentripplanner/otp-react-redux/pull/361/commits/17ff91245c7753225568874d281cf38efe4605df